### PR TITLE
feat(saved-search): Implement saving current search

### PR DESCRIPTION
**Pull request by lynnagara**
Ref: SEN-396

### CODE REVIEW

File: src/sentry/static/sentry/app/views/stream/organizationSavedSearchSelector.jsx [hunk 1]
```diff
@@ -126,13 +140,48 @@ const SaveSearchButton = withApi(
...
+      const {api, organization, onSave} = this.props;
+
       e.preventDefault();
 
-      // TODO: implement saving
+      this.setState({isSaving: true});
+
+      addLoadingMessage(t('Saving Changes'));
+
+      api.request(`/organizations/${organization.slug}/searches/`, {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/stream/organizationSavedSearchSelector.jsx]: Should this use an actionCreator?

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/stream/organizationSavedSearchSelector.jsx]: I'd prefer not to use an action creator when we're not updating a store or any aspect of global state, I think we currently use this pattern more commonly

Comment (markstory) [reply src/sentry/static/sentry/app/views/stream/organizationSavedSearchSelector.jsx]: I was more interested in separating the rendering logic and logic that does API interactions.

There are some tasks in the backlog to move saved-searches into a store, we could move this then.

**Comment by lynnagara** [comment]
@markstory Can you take another look?

**Review by markstory — APPROVED** [review]

**Pull request merged**